### PR TITLE
feat: use helper function to change null to undefined

### PR DESCRIPTION
**Pull request by danielgent**
As requested https://github.com/sesamyab/auth/pull/404#discussion_r1471327693

### CODE REVIEW

File: src/services/token-factory.ts [hunk 1]
```diff
@@ -54,6 +54,19 @@ interface CreateIDTokenParams {
   locale?: string;
 }
 
+function convertNullsToUndefined(payload: IDTokenPayload) {
+  const typeCoercedPayload = payload as any;
```
Comment (danielgent) [inline src/services/token-factory.ts:58]: :see_no_evil: I really don't like this kind of thing

*I'm sure* there must be a typesafe way of doing this. where we can iterate through all the known keys in a type... (although we lose types at runtime of course)


We should also probably be doing this in the getters for the data adapters

_either way_ the tests pass, so this works and we can worry about it later on :clown_face:

Comment (markusahlstrand) [reply src/services/token-factory.ts:58]: Yeah.. agree that the any isn't fun :)

This is chatGpt's suggestion:
```
function removeNullProperties<T extends object>(obj: T): {
  [K in keyof T]: T[K] extends null ? never : T[K]
} {
  const result: any = {};
  Object.keys(obj).forEach((key) => {
    if (obj[key as keyof T] !== null) {
      result[key] = obj[key as keyof T];
    }
  });
  return result;
}
```

I guess it should be possible to the response typed as well.. Tricky

**Review by markusahlstrand — APPROVED** [review]
Think it's better even though it's not really type safe..

File: src/services/token-factory.ts (hunk 1 above)
Comment (danielgent) [reply src/services/token-factory.ts:58]: `const result: any = {};` :smile:   Sounds very chatGPT.  It's the same thing right? an `any` is an `any` :sunglasses:

Comment (danielgent) [reply src/services/token-factory.ts:58]: I tried for a bit... maybe when we both want to practice typescript we can look at this...

I think a typed function that takes specific entities and clears out the `null`s is the safest... But it seems like even Auth0 gets this wrong

**Pull request merged**